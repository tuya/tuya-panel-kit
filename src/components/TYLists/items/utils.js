/* eslint-disable no-param-reassign */

/**
 * https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_pick
 */
export const pick = (object, keys) => {
  return keys.reduce((obj, key) => {
    if (typeof object[key] !== 'undefined') {
      obj[key] = object[key];
    }
    return obj;
  }, {});
};

export const omit = (object, keys) => {
  const shallowCopy = {
    ...object,
  };
  for (let i = 0; i < keys.length; i++) {
    const key = keys[i];
    delete shallowCopy[key];
  }
  return shallowCopy;
};
