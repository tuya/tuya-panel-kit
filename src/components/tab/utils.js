import React from 'react';

const toArray = children => {
  const arr = [];
  React.Children.forEach(children, child => {
    if (child) {
      arr.push(child);
    }
  });
  return arr;
};

const getActiveIndex = (children, activeKey) => {
  const arr = toArray(children);
  for (let i = 0; i < arr.length; i++) {
    if (arr[i].key === activeKey) {
      return i;
    }
  }
  return -1;
};

const activeKeyIsValid = (children, key) => {
  const keys = React.Children.map(children, child => child && child.key);
  return keys.indexOf(key) >= 0;
};

const getDefaultActiveKey = children => {
  let activeKey;
  React.Children.forEach(children, child => {
    if (child && !activeKey && !child.props.disabled) {
      activeKey = child.key;
    }
  });
  return activeKey;
};

export default {
  toArray,
  getActiveIndex,
  activeKeyIsValid,
  getDefaultActiveKey,
};
