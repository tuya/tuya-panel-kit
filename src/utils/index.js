import ColorUtils from './color';
import CoreUtils from './core';
import JsonUtils from './json';
import NumberUtils from './number';
import RatioUtils from './ratio';
import StringUtils from './string';
import TemperatureUtils from './temperature';
import TimeUtils from './time';
import ThemeUtils from './theme';

export {
  ColorUtils,
  CoreUtils,
  JsonUtils,
  NumberUtils,
  RatioUtils,
  StringUtils,
  TemperatureUtils,
  TimeUtils,
  ThemeUtils,
};
