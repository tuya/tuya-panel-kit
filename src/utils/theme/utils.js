import { css } from 'styled-components/native';
import CoreUtils from '../core';

const { get, isNil } = CoreUtils;

export const isObject = item => item && typeof item === 'object' && !Array.isArray(item);

export const deepMerge = (target, ...sources) => {
  if (!sources.length) {
    return target;
  }
  // making sure to not change target (immutable)
  const output = { ...target };
  sources.forEach(source => {
    if (isObject(source)) {
      Object.keys(source).forEach(key => {
        if (isObject(source[key]) && isObject(output[key])) {
          if (!output[key]) {
            output[key] = { ...source[key] };
          } else {
            output[key] = deepMerge(output[key], source[key]);
          }
        } else {
          output[key] = source[key];
        }
      });
    }
  });
  return output;
};

/**
 * @desc 解析上下左右值
 *
 * @param {Array} values - 上右下左值
 * @param {String} key - 样式key，如margin、padding等
 */
export const parseToStyle = (values, key) => {
  if (!values || !Array.isArray(values)) {
    return {};
  }
  const [top, right, bottom, left] = values;
  return {
    [`${key}Top`]: top,
    [`${key}Right`]: right,
    [`${key}Bottom`]: bottom,
    [`${key}Left`]: left,
  };
};

/**
 * @desc 解析上下左右值
 *
 * @param {Array} values - 上右下左值
 * @param {String} key - 样式key，如margin、padding等
 */
export const parseToCss = (values, key) => {
  if (!values || !Array.isArray(values)) {
    return [];
  }
  const [top, right, bottom, left] = values;
  const cssStyle = css`
    ${key}-top: ${top}px;
    ${key}-right: ${right}px;
    ${key}-bottom: ${bottom}px;
    ${key}-left: ${left}px;
  `;
  return cssStyle;
};

/**
 * @desc 从 theme 中取需要的 `theme` 值，
 *
 * @param {Object} props - props，必须包含theme
 * @param {String} key - theme key
 * @param {String | Number} defaultValue - 默认主题值
 *
 * @example
 *
 * 优先级: (以Slider为例)
 * getTheme(props, 'slider.thumbTintColor', '#fff');
 * 
 * props.thumbTintColor ->
 * props.theme.slider.thumbTintColor ->
 * props.theme.slider[props.theme.slider.type].thumbTintColor ->
 * props.theme.slider[props.theme.type].thumbTintColor

 *
 * ```javascript
 *  const StyledView = styled(View)`
 *    background-color: ${props => getTheme(props, 'slider.thumbTintColor', defaultTheme.slider.light.thumbTintColor)};
 * `;
 * ```
 *
 * ```javascript
 *  const defaultTheme = {
 *    type: 'light',
 *    slider: {
 *       type: 'dark',
 *       light: {
 *         thumbTintColor: '#fff',
 *       },
 *       dark: {
 *         thumbTintColor: '#333',
 *         // thumbTintColor: props => props.disabled ? '#999' : '#fff',
 *       }
 *    }
 *  }
 * ```
 */
export const getTheme = (props, key, defaultValue) => {
  const themePath = typeof key === 'function' ? key(props) : key;
  const [namespace, ...path] = key.split('.');
  const themeBasicPath = path.join('.');
  const themeType = get(props.theme, `${namespace}.type`) || get(props.theme, 'type', 'light');
  let themeValue, themeTypePath;
  if (themeType) {
    themeTypePath = [namespace, themeType, ...path].join('.');
  }
  if (!isNil(get(props, themeBasicPath))) {
    themeValue = get(props, themeBasicPath);
  } else if (!isNil(get(props.theme, themePath))) {
    themeValue = get(props.theme, themePath);
  } else {
    themeValue = get(props.theme, themeTypePath, defaultValue);
  }
  if (typeof themeValue === 'function') {
    themeValue = themeValue(props);
  }
  return themeValue;
};

export const createGetTheme = namespace => (key, defaultValue) => {
  const themeKey = `${namespace}.${key}`;
  return getTheme(themeKey, defaultValue);
};
