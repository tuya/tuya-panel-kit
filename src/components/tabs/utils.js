import { RatioUtils, NumberUtils } from '../../utils';
import { PRESS_THRESHOLD, SWIPE_THRESHOLD, VELOCITY_THRESHOLD } from './constant';

const { winWidth } = RatioUtils;
const { inMaxMin } = NumberUtils;

/**
 * @desc 计算tab的宽度
 *
 * @param {Number} maxNum - 一屏上的tab最大数量
 */
export const getTabWidth = (maxNum = 4, tabsWidth = winWidth) => tabsWidth / maxNum;

/**
 * @desc 根据x轴的位置获取当前所处位置tab的索引
 *
 * @param {Number} deltaX - 当前 x 轴的偏移量
 * @param {Number} indexWidth - 一个索引对应的宽度
 */
export const getIndexByDeltaX = (deltaX, indexWidth) => Math.floor(deltaX / indexWidth);

/**
 * @desc 根据x轴偏移量获取距离最近tab的索引
 *
 * @param {Number} deltaX - 当前 x 轴的偏移量
 * @param {Number} indexWidth - 一个索引对应的宽度
 * @param {Number} maxIndex - 最大的索引值
 */
export const getNearestIndexByDeltaX = (deltaX, indexWidth, maxIndex = 100) => {
  if (deltaX > 0) return 0;
  return inMaxMin(0, maxIndex, Math.round(Math.abs(deltaX) / indexWidth));
};

/**
 * @desc 获取索引在循环范围内的邻近的值
 *
 * @param {Number} idx - 新索引
 * @param {Number} min - 最小值
 * @param {Number} max - 最大值
 */
export const getSiblingIndex = (index, min, max) => {
  if (index === min && index === max) {
    return [index, index];
  } else if (index === min) {
    return [index, index + 1];
  } else if (index === max) {
    return [index - 1, index];
  }
  return [index - 1, index + 1];
};

/**
 *
 * @desc 获取居中对齐的tab索引
 *
 * @param {Number} idx - 当前tab的索引
 * @param {Number} oneScreenTabNum - 一屏可放下的tab数量
 * @param {Number} totalNum - tab总数量
 */
export const getCenteredScrollIndex = (idx, oneScreenTabNum, totalNum) => {
  const min = 0;
  const max = Math.max(0, totalNum - oneScreenTabNum);
  const median = Math.ceil(oneScreenTabNum / 2) - 1;
  const scrollToIdx = idx - median;
  const index = inMaxMin(min, max, scrollToIdx);
  return index;
};

/**
 * @desc 判断是否为合法的 Press 事件
 *
 * @param {Number} dx - x轴偏移量
 * @param {Number} dy - y轴偏移量
 */
export const isValidPress = (dx, dy) =>
  Math.abs(dx) <= PRESS_THRESHOLD && Math.abs(dy) <= PRESS_THRESHOLD;

/**
 * @desc 判断是否为合法的 Swipe 事件
 *
 * @param {Number} velocity - 加速度
 * @param {Number} direction - 移动距离
 */
export const isValidSwipe = (velocity, direction) => {
  return Math.abs(velocity) > VELOCITY_THRESHOLD && Math.abs(direction) < SWIPE_THRESHOLD;
};

/**
 * @desc 根据`tabLayout`累加算出每一个`tab`的`left`偏移量
 *
 * @param {Array} tabLayout - tab的布局属性，包含 x, y, width, height
 */
export const reduceTabLayoutLeft = tabLayout => {
  return tabLayout.reduce((acc, cur, index) => {
    if (index === 0) {
      return [{ ...cur, left: cur.x }];
    }
    const prev = acc[index - 1];
    const prevTotalX = prev.left + prev.width + prev.x;
    return [
      ...acc,
      {
        ...cur,
        left: prevTotalX + cur.x,
      },
    ];
  }, {});
};
