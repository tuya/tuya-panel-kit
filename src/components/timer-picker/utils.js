import { CoreUtils } from '../../utils';

const { toFixed } = CoreUtils;

export const getHourSelections = is12Hours => {
  const hours = [];
  for (let i = 0; i < 24; i++) {
    let label = '';
    if (is12Hours) {
      label = i === 0 ? 12 : i <= 12 ? i : i - 12;
    } else {
      label = toFixed(i, 2);
    }
    hours.push({
      value: i,
      label: label.toString(),
    });
  }
  return hours;
};

export const getMinsSelections = () => {
  const minutes = [];
  for (let i = 0; i < 60; i++) {
    minutes.push({
      value: i,
      label: toFixed(i, 2),
    });
  }
  return minutes;
};

export const getTimePrefixSelections = () => [
  {
    value: 'AM',
    label: 'AM',
  },
  {
    value: 'PM',
    label: 'PM',
  },
];

export const getPrefix = hour => (hour >= 12 ? 'PM' : 'AM');
