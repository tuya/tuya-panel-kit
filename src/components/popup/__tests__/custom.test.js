import React from 'react';
import { View, Text } from 'react-native';
import renderer from 'react-test-renderer';
import Popup from '../index';
import _ from 'lodash';
import Custom from '../custom';
import { ThemeUtils } from '../../../utils';

const { ThemeProvider } = ThemeUtils;

describe('Popup Component', () => {
  it('basic render', () => {
    const component = renderer
      .create(
        Popup.custom({
          content: (
            <View
              style={{
                height: 200,
                alignItems: 'center',
                justifyContent: 'center',
                backgroundColor: '#fff',
              }}
            >
              <Text style={{ fontSize: 36, color: '#000' }}>Custom Content</Text>
            </View>
          ),
          title: 'Custom',
          cancelText: '取消',
          confirmText: '确认',
          onMaskPress: ({ close }) => close(),
        })
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});

describe('Custom Component', () => {
  it('basic render', () => {
    const component = renderer.create(
      <Custom
        content={
          <View
            style={{
              height: 200,
              alignItems: 'center',
              justifyContent: 'center',
              backgroundColor: '#fff',
            }}
          >
            <Text style={{ fontSize: 36, color: '#000' }}>Custom Content</Text>
          </View>
        }
        title="自定义"
        showBack={true}
        cancelText="取消"
        confirmText="确认"
        onSwitchValueChange={jest.fn()}
        onMaskPress={jest.fn(({ close: callback }) => callback())}
      />
    );
    const instance = component.getInstance();
    instance._handleSelect();
    instance._handleCancelPress();
    instance._handleBack();
    instance._handleModalClose();
    instance._handleMotionHide();
    instance._handleConfirmPress();
    instance._handleMaskPress();
    instance._handleSwitchValueChange();

    expect(component).toMatchSnapshot();
  });

  it('basic render with visible', () => {
    const component = renderer.create(
      <ThemeProvider>
        <Custom
          content={
            <View
              style={{
                height: 200,
                alignItems: 'center',
                justifyContent: 'center',
                backgroundColor: '#fff',
              }}
            >
              <Text style={{ fontSize: 36, color: '#000' }}>Custom Content</Text>
            </View>
          }
          visible={false}
          title="自定义"
          cancelText="取消"
          showBack={true}
          confirmText="确认"
        />
      </ThemeProvider>
    );

    component.update(
      <ThemeProvider>
        <Custom
          content={
            <View
              style={{
                height: 200,
                alignItems: 'center',
                justifyContent: 'center',
                backgroundColor: '#fff',
              }}
            >
              <Text style={{ fontSize: 36, color: '#000' }}>Custom Content</Text>
            </View>
          }
          visible={true}
          title="自定义"
          showBack={true}
          cancelText="取消"
          confirmText="确认"
        />
      </ThemeProvider>
    );

    expect(component).toMatchSnapshot();
  });
});
