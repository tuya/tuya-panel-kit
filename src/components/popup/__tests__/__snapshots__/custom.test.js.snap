// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Component basic render 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f8f8f8",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "height": 56,
            "justifyContent": "center",
          },
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderTopLeftRadius": 16,
              "borderTopRightRadius": 16,
              "height": 56,
              "justifyContent": "space-around",
              "paddingBottom": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 0,
            },
            Array [
              null,
              "",
            ],
          ],
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 12,
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "height": 36,
                "width": 36,
              },
              false,
              null,
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                false,
                Object {
                  "flex": 0,
                  "height": 36,
                  "width": 36,
                },
              ]
            }
            vbHeight={1024}
            vbWidth={1024}
          >
            <RNSVGPath
              d="M770.673778 21.959111a56.888889 56.888889 0 0 1 0 80.440889l-402.204445 402.318222 402.204445 402.204445a56.888889 56.888889 0 0 1-80.440889 80.497777L247.751111 544.938667a56.888889 56.888889 0 0 1 0-80.497778L690.232889 21.959111a56.888889 56.888889 0 0 1 80.440889 0z"
              fill={
                Array [
                  0,
                  0,
                  0,
                  0,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "fill",
                ]
              }
              stroke={
                Array [
                  0,
                  0,
                  0,
                  0,
                  1,
                ]
              }
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
            />
          </RNSVGSvgView>
        </View>
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color="#000"
          ellipsizeMode="tail"
          size={14}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000",
                "fontSize": 14,
              },
              Array [
                Object {
                  "marginLeft": 2,
                },
                undefined,
              ],
            ]
          }
          text="Back"
          type={null}
          weight={null}
        >
          Back
        </Text>
      </View>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={14}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 14,
            },
            Array [
              Object {
                "color": "rgba(0,0,0,0.9)",
                "fontSize": 14,
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        自定义
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "height": 0.5,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "height": 200,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 36,
          }
        }
      >
        Custom Content
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "height": 56,
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderBottomLeftRadius": 0,
              "borderBottomRightRadius": 0,
              "height": 76,
              "marginTop": 12,
              "paddingBottom": 20,
            },
            null,
          ],
        ]
      }
    >
      <View
        accessibilityLabel="Popup_Custom_Cancel"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 12,
          }
        }
      >
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color={null}
          ellipsizeMode="tail"
          single={false}
          size={16}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "rgba(0,0,0,0.7)",
                  "fontSize": 16,
                },
                null,
              ],
            ]
          }
          type={null}
          weight={null}
        >
          取消
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "height": 24,
              "width": 0.5,
            },
            undefined,
          ]
        }
      />
      <View
        accessibilityLabel="Popup_Custom_Confirm"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "borderRightColor": "rgba(0, 0, 0, 0.1)",
            "borderRightWidth": 0,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 12,
          }
        }
      >
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color={null}
          ellipsizeMode="tail"
          size={16}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#FF4800",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                null,
              ],
            ]
          }
          type={null}
          weight={null}
        >
          确认
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Custom Component basic render with visible 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f8f8f8",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "height": 56,
            "justifyContent": "center",
          },
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderTopLeftRadius": 16,
              "borderTopRightRadius": 16,
              "height": 56,
              "justifyContent": "space-around",
              "paddingBottom": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 0,
            },
            Array [
              null,
              "",
            ],
          ],
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 12,
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "height": 36,
                "width": 36,
              },
              false,
              null,
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                false,
                Object {
                  "flex": 0,
                  "height": 36,
                  "width": 36,
                },
              ]
            }
            vbHeight={1024}
            vbWidth={1024}
          >
            <RNSVGPath
              d="M770.673778 21.959111a56.888889 56.888889 0 0 1 0 80.440889l-402.204445 402.318222 402.204445 402.204445a56.888889 56.888889 0 0 1-80.440889 80.497777L247.751111 544.938667a56.888889 56.888889 0 0 1 0-80.497778L690.232889 21.959111a56.888889 56.888889 0 0 1 80.440889 0z"
              fill={
                Array [
                  0,
                  0,
                  0,
                  0,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "fill",
                ]
              }
              stroke={
                Array [
                  0,
                  0,
                  0,
                  0,
                  1,
                ]
              }
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
            />
          </RNSVGSvgView>
        </View>
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color="#000"
          ellipsizeMode="tail"
          size={14}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000",
                "fontSize": 14,
              },
              Array [
                Object {
                  "marginLeft": 2,
                },
                undefined,
              ],
            ]
          }
          text="Back"
          type={null}
          weight={null}
        >
          Back
        </Text>
      </View>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={14}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 14,
            },
            Array [
              Object {
                "color": "rgba(0,0,0,0.9)",
                "fontSize": 14,
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        自定义
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "height": 0.5,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "height": 200,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 36,
          }
        }
      >
        Custom Content
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "height": 56,
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderBottomLeftRadius": 0,
              "borderBottomRightRadius": 0,
              "height": 76,
              "marginTop": 12,
              "paddingBottom": 20,
            },
            null,
          ],
        ]
      }
    >
      <View
        accessibilityLabel="Popup_Custom_Cancel"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 12,
          }
        }
      >
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color={null}
          ellipsizeMode="tail"
          single={false}
          size={16}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "rgba(0,0,0,0.7)",
                  "fontSize": 16,
                },
                null,
              ],
            ]
          }
          type={null}
          weight={null}
        >
          取消
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "height": 24,
              "width": 0.5,
            },
            undefined,
          ]
        }
      />
      <View
        accessibilityLabel="Popup_Custom_Confirm"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "borderRightColor": "rgba(0, 0, 0, 0.1)",
            "borderRightWidth": 0,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 12,
          }
        }
      >
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color={null}
          ellipsizeMode="tail"
          size={16}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#FF4800",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                null,
              ],
            ]
          }
          type={null}
          weight={null}
        >
          确认
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Popup Component basic render 1`] = `null`;
